### PR TITLE
port ground/compare map filter for perception as segmentation stack

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description
1. move/port **ground filter** to perception stack from sensing stack

- create new directory **object_segmentation** in perception stack
- move ground filter in /perception/object_segmentation **as a package**
- rename package as **ground_segmentation**
- remove unnecessary launch files

2. move/port **compare map filter** to perception stack from sensing stack

- move compare map filter in /perception/object_segmentation **as a package**
- rename package as **compare_map_segmentation**
- remove unnecessary launch files

## Review Procedure

- include guard name
- namespace
- package name

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

**Comment by taichiH** [comment]
Compare map filter is also used for ground filter. 
https://github.com/tier4/AutowareArchitectureProposal.iv/tree/use-autoware-auto-msgs/sensing/preprocessor/pointcloud/pointcloud_preprocessor/include/pointcloud_preprocessor/compare_map_filter

In addiation, compare map filter separetes map and other area segments, so I think it should be included in object_segmentation directory.

**Comment by satoshi-ota** [comment]
I'll rebase after https://github.com/tier4/AutowareArchitectureProposal.iv/pull/507 has been merged.

**Comment by satoshi-ota** [comment]
@taichiH @yukkysaito 
Thank you for your advice :+1: 
I have added not only **ground_filter** but also **compare_map_filter** into object_segmentation directory. So could you please check following points?

- [ ] package name
- [ ] namespace
- [ ] include guard

File: perception/object_segmentation/compare_map_segmentation/package.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  <depend>pointcloud_preprocessor</depend>
+  <depend>rclcpp</depend>
+  <depend>rclcpp_components</depend>
+  <depend>sensor_msgs</depend>
+  <depend>std_msgs</depend>
+  <depend>tf2</depend>
+  <depend>tf2_eigen</depend>
+  <depend>tf2_ros</depend>
+  <depend>tier4_pcl_extensions</depend>
+  <depend>vehicle_info_util</depend>
```
Comment (taichiH) [inline perception/object_segmentation/compare_map_segmentation/package.xml:43]: @satoshi-ota 
I think these are unnecessary dependenciees.
```
-  <depend>autoware_debug_msgs</depend>
-  <depend>autoware_utils</depend>
-  <depend>diagnostic_updater</depend>
-  <depend>image_transport</depend>
-  <depend>lanelet2_extension</depend>
-  <depend>libpcl-all-dev</depend>
-  <depend>message_filters</depend>
-  <depend>nav_msgs</depend>
-  <depend>pcl_conversions</depend>
-  <depend>pcl_msgs</depend>
-  <depend>std_msgs</depend>
-  <depend>tier4_pcl_extensions</depend>
-  <depend>vehicle_info_util</depend>
```

File: perception/object_segmentation/ground_segmentation/package.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  <depend>pointcloud_preprocessor</depend>
+  <depend>rclcpp</depend>
+  <depend>rclcpp_components</depend>
+  <depend>sensor_msgs</depend>
+  <depend>std_msgs</depend>
+  <depend>tf2</depend>
+  <depend>tf2_eigen</depend>
+  <depend>tf2_ros</depend>
+  <depend>tier4_pcl_extensions</depend>
+  <depend>vehicle_info_util</depend>
```
Comment (taichiH) [inline perception/object_segmentation/ground_segmentation/package.xml:43]: Please check dependencies.

**Comment by yukkysaito** [comment]
LGTM

File: perception/object_segmentation/ground_segmentation/package.xml (hunk 2 above)
Comment (satoshi-ota) [reply perception/object_segmentation/ground_segmentation/package.xml]: https://github.com/tier4/AutowareArchitectureProposal.iv/pull/490/commits/06d1758748e0341cba17042e1d1938ef00dabb73 Remove dependencies that are clearly unnecessary.

File: perception/object_segmentation/compare_map_segmentation/package.xml (hunk 1 above)
Comment (satoshi-ota) [reply perception/object_segmentation/compare_map_segmentation/package.xml]: https://github.com/tier4/AutowareArchitectureProposal.iv/pull/490/commits/06d1758748e0341cba17042e1d1938ef00dabb73 Remove dependencies that are clearly unnecessary.

Comment (taichiH) [reply perception/object_segmentation/compare_map_segmentation/package.xml]: Is `pcl_msgs` really needed?

Comment (satoshi-ota) [reply perception/object_segmentation/compare_map_segmentation/package.xml]: Sorry, I have missed it. Thanks :+1:

Comment (satoshi-ota) [reply perception/object_segmentation/compare_map_segmentation/package.xml]: https://github.com/tier4/AutowareArchitectureProposal.iv/pull/490/commits/73811930fb4a1188a5bb89b73006de303135714b please check it again :pray:

File: perception/object_segmentation/ground_segmentation/package.xml (hunk 2 above)
Comment (satoshi-ota) [reply perception/object_segmentation/ground_segmentation/package.xml]: https://github.com/tier4/AutowareArchitectureProposal.iv/pull/490/commits/73811930fb4a1188a5bb89b73006de303135714b please check it again :pray:

File: perception/object_segmentation/compare_map_segmentation/package.xml (hunk 1 above)
Comment (taichiH) [reply perception/object_segmentation/compare_map_segmentation/package.xml]: @satoshi-ota I think `std_msgs` is also unnecessary dependency in compare_map_segmentation pkg.

Comment (taichiH) [reply perception/object_segmentation/compare_map_segmentation/package.xml]: https://github.com/tier4/AutowareArchitectureProposal.iv/pull/490/commits/aa5554d4ba489dc05c59b9a11a0e4bbb0fdda564

LGTM

**Review by taichiH — APPROVED** [review]
LGTM

**Pull request merged**